ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py

```diff
@@ -8,6 +8,59 @@
 from ivy_tests.test_ivy.helpers import handle_cmd_line_args
 
 
+# categorical_crossentropy
+@handle_cmd_line_args
+@given(
+    y_true=st.lists(st.integers(min_value=0, max_value=4), min_size=1, max_size=1),
+    dtype_y_pred=helpers.dtype_and_values(
+        available_dtypes=ivy_tf.valid_float_dtypes,
+        shape=(1, 5),
+        min_value=-10,
+        max_value=10,
+    ),
+    from_logits=st.booleans(),
+    label_smoothing=st.floats(min_value=0, max_value=1),
+    as_variable=st.booleans(),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.tensorflow.categorical_crossentropy"
+    ),
+    native_array=st.booleans()
+)
+def test_categorical_crossentropy(
+    y_true,
+    dtype_y_pred,
+    from_logits,
+    label_smoothing,
+    as_variable,
+    num_positional_args,
+    native_array,
+    fw
+):
+    dtype, y_pred = dtype_y_pred
+    y_pred = ivy.array(y_pred, dtype=ivy.float64)
+    y_true = ivy.array(y_true)
+    ivy.one_hot(y_true, y_pred.shape[-1], out=y_true)
+
+    # Perform softmax on prediction if it's not a probability distribution.
+    if not from_logits:
+        y_pred = ivy.exp(y_pred) / ivy.sum(ivy.exp(y_pred))
```

Comment:
`exp` function doesn't work for `float16` with torch backend and this is giving an error, you will have to fix this.
1 possible fix i can think of is using `assume` in the test function. Its not the most elegant solution.
You can also create a composite function for this since this same logic is being repeated in sparse_categorical_crossentropy
 if you can think of something else feel free to implement it.